### PR TITLE
Mark/stop getting file list

### PR DESCRIPTION
**Pull request by veggiesaurus**
Companion PR of https://github.com/CARTAvis/carta-frontend/pull/1422 and https://github.com/CARTAvis/carta-backend/pull/777

### CODE REVIEW

**Comment by confluence** [comment]
@markccchiang How are you updating `docs/src/defs.rst.txt`, `docs/src/enums.rst.txt` and `docs/src/messages.rst.txt`? These should be updated automatically from the `.proto` files by the makefile in `doc/protoc` (you should just run `make` in that directory to regenerate them). Whenever I do this locally, the files I get are changed; for example, the ordering of the messages is different. Are you perhaps modifying these by hand?

File: shared/enums.proto [hunk 1]
```diff
@@ -67,6 +67,11 @@ enum EventType {
     SPECTRAL_LINE_RESPONSE = 68;
     CONCAT_STOKES_FILES = 69;
     CONCAT_STOKES_FILES_ACK = 70;
+    STOP_FILE_LIST = 71;
```
Comment (veggiesaurus) [inline shared/enums.proto]: @markccchiang can we have a single message with a file type parameter in order to cancel either the image file list or the catalog file list.

File: shared/enums.proto [hunk 2]
```diff
@@ -67,6 +67,11 @@ enum EventType {
...
+    STOP_CATALOG_LIST = 72;
+    FILE_LIST_PROGRESS = 73;
```
Comment (veggiesaurus) [inline shared/enums.proto]: I don't see why we need three separate progress messages that effectively do the same thing. Just make a `FILE_LIST_PROGRESS` event with a `FileListProgress` message which contains a file type parameter.

**Comment by confluence** [comment]
To save time I will push the updated files from here.

**Comment by veggiesaurus** [comment]
@markccchiang please see PR comments

File: docs/src/defs.rst.txt [hunk 3]
```diff
@@ -829,3 +829,33 @@ Source file: `shared/defs.proto <https://github.com/CARTAvis/carta-protobuf/blob
      - 
      - 
 
+.. carta:class:: carta-sub progress
```
Comment (confluence) [inline docs/src/defs.rst.txt]: Please see my comments about updating these files. You should be running `make` in `docs/protoc` to generate them automatically. You need to have `protoc` installed. The Python script used by the makefile automatically sorts these messages by name.

File: docs/src/messages.rst.txt [hunk 4]
```diff
@@ -326,7 +326,11 @@ Source file: `request/catalog_list.proto <https://github.com/CARTAvis/carta-prot
    * - subdirectories
      - string
      - repeated
-     - 
+     -
+   * - cancel
+     - bool
+     -
+     - Whether the Catalog list process is canceled or not
```
Comment (confluence) [inline docs/src/messages.rst.txt]: "Catalog" should not be capitalised

Comment (confluence) [reply docs/src/messages.rst.txt]: A minor point, but "canceled" is the American spelling, and I think we normally use British (which would be "cancelled"). But I'm not 100% sure what we've standardised on. (_Edit: I checked the backend code and we use "cancelled" consistently there._)

File: shared/defs.proto [hunk 5]
```diff
@@ -181,3 +181,9 @@ message Beam {
     float minor_axis = 4;
     float pa = 5;
 }
+
+message Progress {
```
Comment (confluence) [inline shared/defs.proto]: Could we give this a more specific name, and also more documentation in the comments? Out of context it's unclear what this message is for.

File: docs/src/messages.rst.txt (hunk 4 above)
Comment (confluence) [reply docs/src/messages.rst.txt]: These descriptions should go in comments in the proto files; when these docs are generated they are included automatically.

**Review by confluence — CHANGES_REQUESTED** [review]

File: shared/defs.proto (hunk 5 above)
Comment (confluence) [reply shared/defs.proto]: All of these proto file additions should be commented -- the docs are automatically generated from these comments. Any descriptions that are written by hand only into the docs are going to disappear the next time the docs are regenerated.

Comment (confluence) [reply shared/defs.proto]: Perhaps `ListProgress` would be an appropriate generic name. I agree with @angus that there should be only one event type.

**Comment by markccchiang** [comment]
> @markccchiang How are you updating `docs/src/defs.rst.txt`, `docs/src/enums.rst.txt` and `docs/src/messages.rst.txt`? These should be updated automatically from the `.proto` files by the makefile in `doc/protoc` (you should just run `make` in that directory to regenerate them). Whenever I do this locally, the files I get are changed; for example, the ordering of the messages is different. Are you perhaps modifying these by hand?

Thank you for the reminder! I have corrected these text files by automatic generation script.

**Review by confluence — CHANGES_REQUESTED** [review]
This change removes field descriptions from the documentation which were previously added only to the hand-edited `.rst.txt` files and not to the `.proto` files. This applies both to the current changes and to the previous messages added in the Stokes concatenation PR. Please add these as comments to the `.proto` files, so that when the  `.rst.txt` files are regenerated they contain the descriptions.

**Comment by confluence** [comment]
I would actually recommend fixing the Stokes concatenation docs in a separate PR -- splitting out these sets of unrelated changes would make them easier to review. If you are going to do that, please also remove the unnecessary import of `defs.proto` from `control/concat_stokes_files.proto`.

**Comment by confluence** [comment]
The other PR has been merged -- please merge the latest dev into this branch and also re-run the protoc makefile. That should fix the diff.

**Review by confluence — APPROVED** [review]
Looks fine to me.

**Pull request merged**